### PR TITLE
Use repo as of renderctx's member rather than a repoPath on metas

### PR DESCRIPTION
**Pull request by lunny**
Use a `gitrepo.Repository` in the markup's RenderContext but not store the repository's path.

### CODE REVIEW

**Review by wolfogre — APPROVED** [review]

File: modules/markup/markdown/markdown_test.go [hunk 1]
```diff
@@ -222,7 +229,7 @@ See commit 65f1bf27bc
 Ideas and codes
 
 - Bezier widget (by @r-lyeh) ` + AppURL + `ocornut/imgui/issues/786
-- Bezier widget (by @r-lyeh) ` + AppURL + `gogits/gogs/issues/786
+- Bezier widget (by @r-lyeh) ` + AppURL + `user13/repo11/issues/786
```
Comment (yp05327) [inline modules/markup/markdown/markdown_test.go]: Why `AppURL` is used but `testRepoOwnerName` not? 😕 
And shouldn't this be `FullURL`?
Also URLs above

Comment (lunny) [reply modules/markup/markdown/markdown_test.go]: 0815c2910d6e4c8149c44966b12901943ce918a8

File: modules/markup/markdown/markdown_test.go [hunk 2]
```diff
@@ -133,11 +140,11 @@ func testAnswers(baseURLContent, baseURLImages string) []string {
 <li><a href="` + baseURLContent + `/Links" rel="nofollow">Links, Language bindings, Engine bindings</a></li>
 <li><a href="` + baseURLContent + `/Tips" rel="nofollow">Tips</a></li>
 </ul>
-<p>See commit <a href="/gogits/gogs/commit/65f1bf27bc" rel="nofollow"><code>65f1bf27bc</code></a></p>
+<p>See commit <a href="/user13/repo11/commit/65f1bf27bc" rel="nofollow"><code>65f1bf27bc</code></a></p>
```
Comment (yp05327) [inline modules/markup/markdown/markdown_test.go]: Also here

File: modules/markup/markdown/markdown_test.go [hunk 3]
```diff
@@ -133,11 +140,11 @@ func testAnswers(baseURLContent, baseURLImages string) []string {
...
 <p>Ideas and codes</p>
 <ul>
 <li>Bezier widget (by <a href="/r-lyeh" rel="nofollow">@r-lyeh</a>) <a href="http://localhost:3000/ocornut/imgui/issues/786" class="ref-issue" rel="nofollow">ocornut/imgui#786</a></li>
-<li>Bezier widget (by <a href="/r-lyeh" rel="nofollow">@r-lyeh</a>) <a href="http://localhost:3000/gogits/gogs/issues/786" class="ref-issue" rel="nofollow">#786</a></li>
+<li>Bezier widget (by <a href="/r-lyeh" rel="nofollow">@r-lyeh</a>) <a href="http://localhost:3000/user13/repo11/issues/786" class="ref-issue" rel="nofollow">#786</a></li>
```
Comment (yp05327) [inline modules/markup/markdown/markdown_test.go]: Also here

**Review by yp05327 — APPROVED** [review]

**Pull request merged**